src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -271,14 +271,29 @@ boolean isSandboxEnforced(List<Path> services) throws AppEngineException {
   }
 
   /**
-   * Gets the App Engine runtime ID for Java runtimes.
-   *
-   * @param isSandboxEnforced if {@code true}, use Java 8; otherwise, use Java 7
-   * @return "java8" if {@code isJava8} is true; otherwise, returns "java7"
+   * Gets the App Engine runtime ID for Java runtimes. Order of precedence is 11 > 8 > 7. With
+   * default being 7.
    */
   @VisibleForTesting
-  static String getGaeRuntimeJava(boolean isSandboxEnforced) {
-    return isSandboxEnforced ? "java8" : "java7";
+  static String getGaeRuntimeJava(List<Path> services) throws AppEngineException {
+    boolean hasJava8 = false;
+    for (Path serviceDirectory : services) {
+      Path appengineWebXml = serviceDirectory.resolve("WEB-INF/appengine-web.xml");
+      try (InputStream is = Files.newInputStream(appengineWebXml)) {
+        String serviceRuntime = AppEngineDescriptor.parse(is).getRuntime();
+        if (serviceRuntime.startsWith("java11")) {
+          return "java11";
+        } else if (serviceRuntime.startsWith("java8")) {
+          hasJava8 = true;
```

Comment:
Looks like this can be `return "java8";`, and we can remove `hasJava8`.

Reply:
well if the first service is java8 and the second is java11 then we have an issue where java8 shortcuts the search.